__tests__/Ship.test.js

```diff
@@ -0,0 +1,68 @@
+const Ship = require("../src/Ship");
+const Port = require("../src/Port");
+const Itinerary = require("../src/Itinerary");
+
+describe("Ship", () => {
+  describe("with ports and an itinerary", () => {
+    let ship;
+    let dover;
+    let calais;
+    let itinerary;
+
+    beforeEach(() => {
+      dover = {
+        addShip: jest.fn(),
+        removeShip: jest.fn(),
+        name: "Dover",
+        ships: [],
+      };
+
+      calais = {
+        addShip: jest.fn(),
+        removeShip: jest.fn(),
+        name: "Calais",
+        ships: [],
+      };
+
+      itinerary = {
+        ports: [dover, calais],
+      };
+
+      ship = new Ship(itinerary);
+    });
+
+    it("can be instantiated", () => {
+      expect(ship).toBeInstanceOf(Object);
+    });
+
+    it("gets added to port on instantiation", () => {
+      expect(dover.addShip).toHaveBeenCalledWith(ship);
+    });
+
+    it("has a starting port", () => {
+      expect(ship.currentPort).toBe(dover);
+    });
```

Comment:
Should add tests for all instantiated values, i.e. add a test to check the `previousPort` is instantiated as `null`